packages/firestore/lite/src/api/reference.ts

```diff
@@ -306,6 +307,15 @@ export function getDoc<T>(
   });
 }
 
+export function setDoc<T>(
+  reference: firestore.DocumentReference<T>,
+  data: T
+): Promise<void>;
+export function setDoc<T>(
+  reference: firestore.DocumentReference<T>,
+  data: Partial<T>,
+  options: firestore.SetOptions
+): Promise<void>;
```

Comment:
Adding these overloads allows us to simulator more realtime usage in the integration tests as it removes the availability of the "catch all" API.